platform/index.js

```diff
@@ -90,6 +94,66 @@ function getFilesDir() {
         return path.resolve(getUserConfigDir(), 'almond-cmdline');
 }
 
+class CmdlineThingpediaClient extends Tp.HttpClient {
+    constructor(platform) {
+        super(platform, process.env.THINGPEDIA_URL || Config.THINGPEDIA_URL);
+    }
+
+    async _getLocalDeviceManifest(manifestPath, deviceKind) {
+        const ourMetadata = (await util.promisify(fs.readFile)(manifestPath)).toString();
+        const ourParsed = ThingTalk.Grammar.parse(ourMetadata);
+        ourParsed.classes[0].annotations.version = new ThingTalk.Ast.Value.Number(-1);
+
+        if (!ourParsed.classes[0].is_abstract) {
+            try {
+                // ourMetadata might lack some of the fields that are in the
+                // real metadata, such as api keys and OAuth secrets
+                // for that reason we fetch the metadata for thingpedia as well,
+                // and fill in any missing parameter
+                const officialMetadata = await super.getDeviceCode(deviceKind);
```

Comment:
so, if the device is not on thingpedia yet, it will throw an error?

Reply:
And then catch it a few lines later, yes.

Reply:
I see, 404 error is catched. But still, we should probably check if undefined is in config first to avoid unnecessary fetch.

Reply:
I suppose we could? This is the same code as thingpedia-common-devices btw (and we should share it)